packages/crypto/src/utils.ts

```diff
@@ -63,6 +65,54 @@ export function checkValidProperty(params: {
   }
 }
 
+/**
+ * Determines the JOSE algorithm identifier of the digital signature algorithm based on the `alg` or
+ * `crv` property of a {@link Jwk | JWK}.
+ *
+ * If the `alg` property is present, its value takes precedence and is returned. Otherwise, the
+ * `crv` property is used to determine the algorithm.
+ *
+ * @see {@link https://www.iana.org/assignments/jose/jose.xhtml#web-signature-encryption-algorithms | JOSE Algorithms}
+ * @see {@link https://datatracker.ietf.org/doc/draft-ietf-jose-fully-specified-algorithms/ | Fully-Specified Algorithms for JOSE and COSE}
+ *
+ * @example
+ * ```ts
+ * const publicKey: Jwk = {
+ *   "kty": "OKP",
+ *   "crv": "Ed25519",
+ *   "x": "FEJG7OakZi500EydXxuE8uMc8uaAzEJkmQeG8khXANw"
+ * }
+ * const algorithm = getJoseSignatureAlgorithmFromPublicKey(publicKey);
+ * console.log(algorithm); // Output: "EdDSA"
+ * ```
+ *
+ * @param publicKey - A JWK containing the `alg` and/or `crv` properties.
+ * @returns The name of the algorithm associated with the key.
+ * @throws Error if the algorithm cannot be determined from the provided input.
+ */
+export function getJoseSignatureAlgorithmFromPublicKey(publicKey: Jwk): string {
+  const curveToJoseAlgorithm: Record<string, string> = {
```

Comment:
typescript q: is `Record` better than `Map` in this context?

Reply:
Good question.  There are performance advantages and you do get nicer lookups but slightly more verbose initialization and reverse lookups.  

But practically speaking, this data structure is so small (and will remain so given how few JOSE DSAs exist) that either approach is acceptable.  I opted to take the slightly more terse route in this case.

Reply:
@jiyoontbd Additional typescript neckbeardery if you're curious / didn't already know.

`Record` is pure typescript in the sense that javascript does not have a notion of `Record`. A `Record` is just an plain old javascript object, but with some detail provided for the typescript compiler.

`Map` exists in javascript at runtime, so it's not "just" a typescript thing. `Map` is a [class that you can instantiate directly](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) using `new Map()` and use instance methods `.get()` or `.set()`.

Like Frank mentioned, the difference in performance at runtime between a plain old javascript object and a `Map` is negligible for our purposes. [This stackoverflow answer](https://stackoverflow.com/questions/18541940/map-vs-object-in-javascript) provides some color on when `Map` is more useful.